lib/file_transfer_agent/encrypt_util.js

```diff
@@ -32,6 +35,62 @@ function EncryptionMetadata(key, iv, matDesc) {
 
 exports.EncryptionMetadata = EncryptionMetadata;
 
+function TempFileGenerator() {
+
+  this.fileSync = function (option = { dir: os.tmpdir(), prefix: '', postfix: '', extension: '' }) {
+    const randomName = crypto.randomUUID();
+    const fileName = `${option.prefix || ''}${randomName}${option.postfix || ''}${option.extension || ''}`;
+      
+    if (!this.checkDirInTemp(option.dir)) {
+      option.dir = os.tmpdir();
+    }
+
+    const fullpath = path.join( option.dir, fileName);
+    fs.writeFileSync(fullpath, '');
+
+    const fileDescripter = fs.openSync(fullpath);
+    return { name: fullpath, fd: fileDescripter };
+  };
+  
+  this.file = function (option = { dir: os.tmpdir(), prefix: '', postfix: '', extension: '' }, callback) {
+    let error;
+    const randomName = crypto.randomUUID();
+    const fileName = `${option.prefix || ''}${randomName}${option.postfix || ''}${option.extension || ''}`;
+
+    if (!this.checkDirInTemp(option.dir)) {
+      option.dir = os.tmpdir();
+    }
+
+    const fullpath = path.join( option.dir, fileName);
+    
+    try {
+      fs.writeFileSync(fullpath, '');
```

Comment:
I'm not familiar with node.js and always get confused with sync/async functions. Is it OK to call sync functions in async function? Would it be better to return a promise?

Reply:
It's ok to call sync functions in an async function. However, it would be preferred to use async functions

But this function doesn't have an async modifier though `this.file = function`

Reply:
we have both this.fileSync and this.file so I think this.file supposed to be async function?

Reply:
It's the callback version of `fileSync`

Reply:
sounds good.